src/logic/import/voc/VOCImporter.ts

```diff
@@ -0,0 +1,105 @@
+import {ImageData, LabelName, LabelRect} from '../../../store/labels/types';
+import {LabelUtil} from "../../../utils/LabelUtil";
+import {AnnotationImporter} from '../AnnotationImporter';
+import {LabelsSelector} from '../../../store/selectors/LabelsSelector';
+
+type FileParseResult = {
+    filename: string,
+    labeledBoxes: LabelRect[]
+};
+
+type VOCImportResult = {
+    labelNames: Record<string, LabelName>,
+    fileParseResults: FileParseResult[],
+};
+
+export class VOCImporter extends AnnotationImporter {
+    public import(
+        filesData: File[],
+        onSuccess: (imagesData: ImageData[], labelNames: LabelName[]) => any,
+        onFailure: (error?:Error) => any
+    ): void {
+        try {
+            const inputImagesData: Record<string, ImageData> = VOCImporter.mapImageData();
+
+            this.loadAndParseFiles(filesData).then(results => {
+                for (const result of results.fileParseResults) {
+                    if (inputImagesData[result.filename]) {
+                        inputImagesData[result.filename].labelRects = result.labeledBoxes;
+                    }
+                }
+
+                onSuccess(
+                    Array.from(Object.values(inputImagesData)),
+                    Array.from(Object.values(results.labelNames))
+                );
+            }).catch((error: Error) => onFailure(error));
+        } catch (error) {
+            onFailure(error as Error)
+        }
+    }
+
+    private loadAndParseFiles(files: File[]): Promise<VOCImportResult> {
+        const parser = new DOMParser();
+
+        return Promise.all(files.map(file => file.text())).then(textFiles => 
```

Comment:
Could we add typing for `file`, `textFiles`, `current` and `fileData`? I know it may seem unnecessary, but I really love to have types especially when I'm using a function chaingn. It is much easier to understand what is happening especially after some time.